src/ORM/DataList.php

```diff
@@ -1749,20 +1727,18 @@ public function byIDs($ids)
      * Return the first DataObject with the given ID
      *
      * The object returned is not cached, unlike {@link DataObject::get_by_id()}
+     *
      * @return T|null
      */
-    public function byID($id)
+    public function byID(int|string|null $id): ?DataObject
```

Comment:
Being pragmatic, running this through kitchen-sink there was of existing code that will pass in integers as strings, and null.

I put an inline comment on SS_List as to why I did this

It should by only int, including on SS_List, there's a docblock type of $int on Filterable::byID()

While we could fix up all our own references to this, it's a heavily used method and will break a lot of project s e.g. `$id = $request->param('ID');` will probably be a string

I'd by OK removing the int|string|null typehint on SS_List and just having it as an docblock $int instead

Reply:
The point of strong typing is to clean up inconsistencies like that.
If we're adding a strong typehint here, we should clean up the relevant core code so that it passes the correct typed value into the method.

The alternative is to just not include a type here, rather than saying "it can be a string or null or an int but should be int but isn't always int"

Reply:
Changed to mixed

Reply:
Shouldn't be mixed for DataList but whatever